plugins/pulp_docker/plugins/migrations/0006_add_arch_os_size_fields.py

```diff
@@ -0,0 +1,58 @@
+import logging
+import os
+
+from pulp.server.db.connection import get_collection
+from pulp.server.db.migrations.lib import utils
+
+
+_logger = logging.getLogger('pulp_docjer.plugins.migrations.0006')
+
+
+def migrate(*args, **kwargs):
+    migrate_manifest_list()
+    migrate_blob()
+
+
+def migrate_manifest_list():
+    """
+    Update format of manifest on manifest lists to have digest, os, and arch fields
+    """
+
+    manifest_list_collection = get_collection('units_docker_manifest_list')
+    total_manifest_lists = manifest_list_collection.count(
+        {'manifests': {'$type': ["string"]}})
+
+    with utils.MigrationProgressLog('Update manifest field on ManifestList',
+                                    total_manifest_lists) as migration_log:
+        for manifest_list in manifest_list_collection.find(
+                {'manifests': {'$type': ["string"]}}).batch_size(100):
+
+            manifests = []
+            for digest in manifest_list['manifests']:
+
+                manifest = {'digest': digest,
+                            'os': '',
+                            'arch': ''}
+                manifests.append(manifest)
+
+            manifest_list_collection.update(
+                {'_id': manifest_list['_id']},
+                {'$set': {'manifests': manifests}})
+            migration_log.progress()
+
+
+def migrate_blob():
+    """
+    Add size field to blobs
+    """
+    blob_collection = get_collection('units_docker_blob')
+
+    total_blobs = blob_collection.count({'size': {'$exists': False}})
+    with utils.MigrationProgressLog('Update size on Blob', total_blobs) as migration_log:
+        for blob in blob_collection.find({'size': {'$exists': False}}).batch_size(100):
+            blob_collection.update(
+                {'_id': blob['_id']},
+                {'$set': {'size': os.path.getsize(blob['_storage_path'])}}
```

Comment:
if i am not mistaken it was agreed to retrieve this info from manifest.json and keep it on the Manifest model in the FSLayers. This way the size of the blob would be visible trough the 1 api query to the manifest collection.
Otherwise with this approach 2 calls would need to be made to see the size of each blob present within the given X manifest.
@dkliban can you confirm my understanding?

Reply:
The last comment stated that size should be stored on the blob model: https://pulp.plan.io/issues/3137#note-13

Reply:
yes, but then when the description wsa updated that got changed https://pulp.plan.io/journals/29125/diff?detail_id=29792

Reply:
Sorry about all the confusion :(